django-backend/fecfiler/committee_accounts/utils.py

```diff
@@ -90,37 +76,129 @@ def create_committee_account(committee_id, user):
-    # To be verified in https://fecgov.atlassian.net/browse/FECFILE-1706
-    committee_data = query_production_efo(committee_id)
+"""
+FEC API methods
+"""
+
+
+def query_fec_api(endpoint, params):
+    """Shared method to query an EFO API"""
+    headers = {"Content-Type": "application/json"}
+    response = requests.get(endpoint, headers=headers, params=params)
+    response_data = response.json()
+    committee_results = response_data.get("results", [])
+    return committee_results[0] if committee_results else None
+
+
+"""
+Production FEC
+"""
+
+
+def get_production_committee_emails(committee_id):
+    """
+    First query the raw endpoint,
+    if no raw data is available, query the processed endpoint
+    """
+    committee_data = get_raw_committee_data(committee_id)
+
+    if committee_data is None:
+        committee_data = get_processed_committee_data(committee_id)
+
+    return committee_data.get("email", None) if committee_data else None
+
+
+def get_production_committee_data(committee_id):
+    """
+    First query the processed endpoint,
+    if no processed data is available, query the raw endpoint
+    """
+    # first try processed endpoint
+    committee_data = get_processed_committee_data(committee_id)
+
     if committee_data is None:
-        return None
+        # if no processed data, try raw endpoint
+        committee_data = get_raw_committee_data(committee_id)
+
+    return committee_data
+
 
-    # Committee Type Label
-    committee_data["committee_type_label"] = committee_data.get(
-        "committee_type_full", None
+def get_processed_committee_data(committee_id):
+
+    params = {
+        "api_key": settings.PRODUCTION_OPEN_FEC_API_KEY,
+        "committee_id": committee_id,
+    }
+    committee_data = query_fec_api(
+        f"{settings.PRODUCTION_OPEN_FEC_API}committee/{committee_id}/", params
     )
 
-    # PAC/PTY
-    committee_data["isPTY"] = is_production_efo_pty(committee_data)
-    committee_data["isPAC"] = is_production_efo_pac(committee_data)
+    if committee_data:
+        # Committee Type Label
+        committee_data["committee_type_label"] = committee_data.get(
+            "committee_type_full", None
+        )
+
+        # PAC/PTY
+        committee_data["isPTY"] = is_production_efo_pty(committee_data)
+        committee_data["isPAC"] = is_production_efo_pac(committee_data)
+
+        # Qualified
+        committee_data["qualified"] = (
+            committee_data.get("committee_type") in PRODUCTION_QUALIFIED_COMMITTEES
+        )
+
+    return committee_data
+
 
-    # Qualified
-    committee_data["qualified"] = (
-        committee_data.get("committee_type") in PRODUCTION_QUALIFIED_COMMITTEES
+def get_raw_committee_data(committee_id):
+    params = {
+        "api_key": settings.PRODUCTION_OPEN_FEC_API_KEY,
+        "committee_id": committee_id,
+    }
+    committee_data = query_fec_api(
+        f"{settings.PRODUCTION_OPEN_FEC_API}efile/form1/", params
     )
 
+    if committee_data:
```

Comment:
is it worth factoring out this section into a shared function to be used here and in get_committee_from_test_fec() ?

Reply:
good point i was thinking the same thing, but was betting that there would be differences that would force us to split them up again. After speaking with Sasha this afternoon, it looks like there will be differences in this snippet from the test one.